functions/src/resource.ts

```diff
@@ -36,7 +39,12 @@ export class BaseResourceObject implements BaseResource {
     return {id, name, description, type, tool, accessRules};
```

Comment:
This seems to be the place where the accessRules for the resource would be returned in a get request. So it seems the get request needs to be authenticated otherwise it will expose the readWrite token.  I think this might conflict with some other usage of the get request, so perhaps it requires more discussion.